test/e2e/lib/helpers/browser.ts

```diff
@@ -56,8 +56,9 @@ export async function sendXhr(url: string, headers: string[][] = []): Promise<st
   const result: State = await browserExecuteAsync(
     (url, headers, done) => {
       const xhr = new XMLHttpRequest()
-      xhr.addEventListener('load', () => done({ state: 'success', response: xhr.response as string }))
-      xhr.addEventListener('error', () => done({ state: 'error' }))
+      let state: State = { state: 'error' }
+      xhr.addEventListener('load', () => (state = { state: 'success', response: xhr.response as string }))
```

Comment:
NIT: I find this more readable:

```suggestion
      xhr.addEventListener('load', () => {
        state = { state: 'success', response: xhr.response as string }
      })
```